### PR TITLE
[GEOS-9678] SldService numberFormatException if percentages true with…

### PR DESCRIPTION
**Pull request by taba90**
… customClasses

Jira ticket https://osgeo-org.atlassian.net/browse/GEOS-9678

This pr fixes a bug with percentages parameter in sldService when customClasses are specifying intervals not matching data.

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

**Submitting the PR does not require you to check all items, but by the time it gets merged, they should be either satisfied or inapplicable.**

For all pull requests:

- [x] Confirm you have read the [contribution guidelines](https://github.com/geoserver/geoserver/blob/master/CONTRIBUTING.md) 
- [x] You have sent a Contribution Licence Agreement (CLA) as necessary (not required for small changes, e.g., fixing typos in documentation)
- [x] Make sure the first PR targets the master branch, eventual backports will be managed later. This can be ignored if the PR is fixing an issue that only happens in a specific branch, but not in newer ones.

The following are required only for core and extension modules (they are welcomed, but not required, for community modules):
- [x] There is a ticket in Jira describing the issue/improvement/feature (a notable exemptions is, changes not visible to end users)
- [x] PR for bug fixes and small new features are presented as a single commit
- [x] Commit message must be in the form "[GEOS-XYZW] Title of the Jira ticket" (export to XML in Jira generates the message in this exact form)
- [x] The pull request contains changes related to a single objective. If multiple focuses cannot be avoided, each one is in its own commit and has a separate ticket describing it.
- [x] New unit tests have been added covering the changes
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] This PR passes the [QA checks](https://docs.geoserver.org/latest/en/developer/qa-guide/index.html) (QA checks results will be reported by travis-ci after opening this PR)
- [ ] Commits changing the UI, existing user workflows, or adding new functionality, need to include documentation updates (screenshots, text)
- [ ] Commits changing the REST API, or any configuration object, should check if the REST API docs (Swagger YAML files and classic documentation) need to be updated.

### CODE REVIEW

File: src/extension/sldService/src/main/java/org/geoserver/sldservice/utils/classifier/RasterSymbolizerBuilder.java [hunk 1]
```diff
@@ -496,7 +496,9 @@ private String getPercentagesLabelPortion(double[] percentages, int i) {
         double total = DoubleStream.of(classMembers).sum();
         double[] percentages = new double[classNum];
         for (int i = 0; i < classNum; i++) {
-            percentages[i] = (classMembers[i] / total) * 100;
+            double cm = classMembers[i];
+            if (cm != 0d && total != 0d) percentages[i] = (classMembers[i] / total) * 100;
```
Comment (aaime) [inline src/extension/sldService/src/main/java/org/geoserver/sldservice/utils/classifier/RasterSymbolizerBuilder.java:500]: You extrac cm, but then use classMembers[i] right next line? :-D

```suggestion
            if (cm != 0d && total != 0d) percentages[i] = (cm / total) * 100;
```

File: src/extension/sldService/src/main/java/org/geoserver/sldservice/utils/classifier/RasterSymbolizerBuilder.java [hunk 2]
```diff
@@ -509,7 +511,9 @@ private String getPercentagesLabelPortion(double[] percentages, int i) {
         double[] percentages = new double[intervals];
         int total = IntStream.of(bins).sum();
         for (int i = 0; i < intervals; i++) {
-            percentages[i] = ((double) bins[i] / total) * 100;
+            double classMembers = bins[i];
```
Comment (aaime) [inline src/extension/sldService/src/main/java/org/geoserver/sldservice/utils/classifier/RasterSymbolizerBuilder.java:514]: cm or classMembers? Consistency please :-D

File: src/extension/sldService/src/main/java/org/geoserver/sldservice/utils/classifier/RulesBuilder.java [hunk 3]
```diff
@@ -559,7 +559,10 @@ private Expression normalizeProperty(
     private double[] computeCustomPercentages(int[][] bins, double totalSize) {
         double[] percentages = new double[bins.length];
         for (int i = 0; i < bins.length; i++) {
-            percentages[i] = ((double) bins[i][0] / totalSize) * 100;
+            double classMembers = bins[i][0];
+            if (classMembers != 0d && totalSize != 0d)
+                percentages[i] = ((double) bins[i][0] / totalSize) * 100;
```
Comment (aaime) [inline src/extension/sldService/src/main/java/org/geoserver/sldservice/utils/classifier/RulesBuilder.java:564]: See above

**Comment by taba90** [comment]
thanks @aaime your feedbacks have been applied

**Comment by aaime** [comment]
Merged, backport as you see fit.

**Pull request merged**